src/NuGet.Core/NuGet.Packaging/Rules/TargetFrameworkMissingPlatformVersionRule.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    internal class TargetFrameworkMissingPlatformVersionRule : IPackageRule
+    {
+        // NOTE: We never actually generate messages, only error on failure.
+        public string MessageFormat => "";
+
+        public TargetFrameworkMissingPlatformVersionRule()
+        {
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            return Validate(new NuspecReader(builder.GetNuspec()), builder.GetFiles());
```

Comment:
Why do you need a new NuspecReader object?  Can you use the existing one instead ?

https://github.com/NuGet/NuGet.Client/blob/087a7113ad70ae4d4c538eba0f387271af0c649b/src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs#L112

```suggestion
            return Validate(builder.NuspecReader, builder.GetFiles());
```